src/findings_data_import.py

```diff
@@ -132,103 +408,35 @@ def import_data(
+            ssm_db_user=ssm_db_user,
+            ssm_db_password=ssm_db_password,
+            db_hostname=db_hostname,
+            db_port=db_port,
         )
 
-        # Connect to MongoDB with timeout so Lambda doesn't run over
-        db_connection = MongoClient(
-            host=db_uri, serverSelectionTimeoutMS=2500, tz_aware=True
+        logging.info("Extracting/validating findings from %s", data_filename)
+        valid_findings = extract_findings(
+            findings_data=findings_data, field_map_dict=field_map_dict
         )
-        db = db_connection[db_info["db_name"]]
-        logging.info(
-            "DB connection set up to %s:%s/%s", db_hostname, db_port, db_info["db_name"]
-        )
-
-        processed_findings = 0
-        # Iterate through data and save each record to the database
-        for index, finding in enumerate(findings_data):
-            # Replace or rename fields from replacement JSON
-            for field in field_map_dict:
-                if field in finding.keys():
-                    if field_map_dict[field]:
-                        finding[field_map_dict[field]] = finding[field]
-                    finding.pop(field, None)
-
-            # Get RVA ID in format DDDD([.-]D+) from the end of the "RVA ID" field.
-            rvaId = re.search(r"(\d{4})(?:[.-](\d+))?$", finding["RVA ID"])
-            if rvaId:
-                rID = f"RV{rvaId.group(1)}"
-                if rvaId.group(2) is not None:
-                    rID += f".{rvaId.group(2)}"
-                finding["RVA ID"] = rID
-            else:
-                skip_record(
-                    index,
-                    data_filename,
-                    'Unable to extract valid RVA ID from "%s"',
-                    finding["RVA ID"],
-                )
-                continue
-            # Only process appropriate findings records.
-            if "RVA ID" in finding.keys() and "NCATS ID" in finding.keys():
-                # If the finding already exists, update it with new data.
-                # Otherwise insert it as a new document (upsert=True).
-                db.findings.find_one_and_update(
-                    {
-                        "RVA ID": finding["RVA ID"],
-                        "NCATS ID": finding["NCATS ID"],
-                        "Severity": finding["Severity"],
-                    },
-                    {"$set": finding},
-                    upsert=True,
-                )
-
-                processed_findings += 1
-            else:
-                skip_record(
-                    index, data_filename, 'Missing "RVA ID" or "NCATS ID" field'
-                )
+        logging.info("Updating records")
+        for finding in valid_findings:
+            update_record(db=db, finding=finding)
 
         logging.info(
-            '%d/%d documents successfully processed from "%s"',
-            processed_findings,
+            "%s/%s documents successfully processed from '%s'.",
```

Comment:
I think the first two variables are always `int`s:
```suggestion
            "%d/%d documents successfully processed from '%s'.",
```